### PR TITLE
JSON Grains - Add test case to validate grains using JSON JObject parameters

### PR DESCRIPTION
**Pull request by jthelin**
- A test case for grains that use JSON JObject parameters for input / output.

- See #1368 for some related background discussion.

### CODE REVIEW

**Comment by jthelin** [comment]
For backgrounds interest, did you know there are many subtle variations to the "standard" JSON serialization format?

https://issues.apache.org/jira/browse/SPARK-11745

Thankfully I believe.NET app developers can pretty much rely on NewtonSoft.Json library to provide abstraction layer over those many edge cases and degrees of optionality.

File: src/TestGrainInterfaces/TestGrainInterfaces.csproj.rej [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff a/src/TestGrainInterfaces/TestGrainInterfaces.csproj b/src/TestGrainInterfaces/TestGrainInterfaces.csproj	(rejected hunks)
```
Comment (sergeybykov) [inline src/TestGrainInterfaces/TestGrainInterfaces.csproj.rej]: This is garbage from merge accidentally included?

Comment (jthelin) [inline src/TestGrainInterfaces/TestGrainInterfaces.csproj.rej]: Yes indeed. Removed.

**Comment by jthelin** [comment]
Removed the .rej junk file and rebased against master 7350accdccefa31e146c6a2f5b3ce479ff330d70

**Pull request merged**